common/extended_window.hpp

```diff
@@ -43,89 +48,105 @@ namespace ILLIXR{
-            {
-                XVisualInfo *vi = glXGetVisualFromFBConfig( dpy, fbc[i] );
-                if ( vi )
-                {
+            for (i = 0; i < fbcount; ++i) {
+                XVisualInfo *vi = glXGetVisualFromFBConfig(dpy, fbc[i]);
+                if (vi) {
                     int samp_buf, samples;
-                    glXGetFBConfigAttrib( dpy, fbc[i], GLX_SAMPLE_BUFFERS, &samp_buf );
-                    glXGetFBConfigAttrib( dpy, fbc[i], GLX_SAMPLES       , &samples  );
+                    glXGetFBConfigAttrib(dpy, fbc[i], GLX_SAMPLE_BUFFERS, &samp_buf);
+                    glXGetFBConfigAttrib(dpy, fbc[i], GLX_SAMPLES       , &samples );
 #ifndef NDEBUG
-                    printf( "  Matching fbconfig %d, visual ID 0x%2lx: SAMPLE_BUFFERS = %d,"
-                            " SAMPLES = %d\n",
-                            i, vi -> visualid, samp_buf, samples );
+                    printf("  Matching fbconfig %d, visual ID 0x%2lx: SAMPLE_BUFFERS = %d,"
+                           " SAMPLES = %d\n",
+                           i, vi -> visualid, samp_buf, samples);
 #endif
-                    if ( best_fbc < 0 || (samp_buf && samples > best_num_samp) )
+                    if (best_fbc < 0 || (samp_buf && samples > best_num_samp))
                       best_fbc = i, best_num_samp = samples;
-                    if ( worst_fbc < 0 || !samp_buf || samples < worst_num_samp )
+                    if (worst_fbc < 0 || !samp_buf || samples < worst_num_samp)
                       worst_fbc = i, worst_num_samp = samples;
                 }
-                XFree( vi );
+                XFree(vi);
             }
-            GLXFBConfig bestFbc = fbc[ best_fbc ];
-            // Be sure to free the FBConfig list allocated by glXChooseFBConfig()
-            XFree( fbc );
+
+            GLXFBConfig bestFbc = fbc[best_fbc];
+
+            // Free the FBConfig list allocated by glXChooseFBConfig()
+            XFree(fbc);
+
             // Get a visual
-            XVisualInfo *vi = glXGetVisualFromFBConfig( dpy, bestFbc );
+            XVisualInfo *vi = glXGetVisualFromFBConfig(dpy, bestFbc);
 #ifndef NDEBUG
-            printf( "Chosen visual ID = 0x%lx\n", vi->visualid );
-#endif
+            printf("Chose visual ID = 0x%lx\n", vi->visualid);
 
+            printf("Creating colormap\n");
+#endif
             Colormap cmap = XCreateColormap(dpy, root, vi->visual, AllocNone);
-            XSetWindowAttributes swa;
-            swa.colormap = cmap;
-            swa.event_mask = ExposureMask | KeyPressMask;
-            win = XCreateWindow(dpy, root, 0, 0, width, height, 0, vi->depth, InputOutput, vi->visual, CWColormap | CWEventMask, &swa);
-            XMapWindow(dpy, win);
+
+#ifndef NDEBUG
+            printf("Creating window\n");
+#endif
+            XSetWindowAttributes attributes;
+            attributes.colormap = cmap;
+            attributes.background_pixel = 0;
+            attributes.border_pixel = 0;
```

Comment:
Are these changes necessary for headless or something else?

Reply:
This is _the_ fix for headless.